packages/block-editor-tools/src/components/post-picker/post-list.tsx

```diff
@@ -0,0 +1,205 @@
+   */
+  const getPosts = useCallback((params: Params, cancelled: Boolean = false) => {
+    /**
+     * Gets the api path.
+     *
+     * @param {*} params The parameters.
+     */
+    function getPath() {
+      let path = addQueryArgs(
+        baseUrl,
+        {
+          page: params.page,
+        },
+      );
+      if (params.searchValue && params.searchValue.length > 2) {
+        path = addQueryArgs(
+          path,
+          {
+            search: params.searchValue,
+          },
+        );
+      }
+      return path;
+    }
+
+    if (params.searchValue && params.searchValue.length <= 2) {
+      return;
+    }
+    const path = getPath();
+    setIsUpdating(true);
+    apiFetch({ path, parse: false }).then((response) => {
+      setTotalPages(parseInt(
+        // @ts-ignore
+        response.headers.get('X-WP-TotalPages'),
+        10,
+      ));
+      // @ts-ignore
+      return response.json();
+    }).then((result) => {
+      let posts = result as any as WP_REST_API_Search_Results;
+      if (params.page > 1) {
+        posts = [
+          ...listposts,
+          ...result as any as WP_REST_API_Search_Results,
+        ];
+      }
+      if (cancelled) {
+        return;
+      }
+      // @ts-ignore
+      setListposts(posts as any as WP_REST_API_Search_Results);
+      setIsUpdating(false);
+    });
+  }, [listposts, baseUrl]);
+
+  /**
+   * Loads more posts.
+   */
+  function loadMore() {
+    const newParams = {
+      ...pathParams,
+      page: pathParams.page + 1,
+    };
+    setPathParams(newParams);
+    getPosts(newParams);
+  }
+
+  /**
+   * Handles a change to the search text string.
+   * @param {event} event - The event from typing in the text box.
+   */
+  function handleSearchTextChange(event: Event) {
+    const {
+      target: {
+        // @ts-ignore
+        value = '',
+      } = {},
+    } = event;
+
+    const newParams = {
+      ...pathParams,
+      searchValue: value,
+      page: 1,
+    };
+    setPathParams(newParams);
+    getPosts(newParams);
+  }
+
+  // Load posts on page load.
+  useEffect(() => {
+    let cancelled = false;
+    if (!initialLoad) {
+      setInitialLoad(true);
+      getPosts(pathParams, cancelled);
+    }
+    return () => {
+      cancelled = true;
+    };
+  }, [getPosts, initialLoad, pathParams]);
+
+  return (
+    <>
+      <input
+        type="text"
+        value={pathParams.searchValue}
+        placeholder={__('Search...', 'alley-scripts')}
+        label={__('Search', 'alley-scripts')}
+        // @ts-ignore
+        onChange={handleSearchTextChange}
+      />
+      <div className="alley-scripts-post-picker__post-list">
+        {listposts ? (
+          listposts.map((t) => (
+            <Button
+              key={t.id}
+              className={classNames({
+                'alley-scripts-post-picker__post': true,
+                'is-selected': t.id === selected,
+              })}
+              onClick={() => setSelected(t.id)}
```

Comment:
We can fix the string/number problem here:
```suggestion
              onClick={() => setSelected(parseInt(t.id, 10))}
```